app/config/set_parameters.php

```diff
@@ -55,7 +55,26 @@
         $container->setParameter($key, $value);
     }
 
-    $container->setParameter('cache.driver', extension_loaded('apc') ? 'apc': 'array');
+    $driver = 'array';
+    $cacheType = [
+        'CacheMemcache' => ['memcache'],
+        'CacheMemcached' => ['memcached'],
+        'CacheApc' => ['apcu', 'apc'],
+        'CacheXcache' => ['xcache'],
+    ];
+
+    if (isset($parameters['parameters']['ps_cache_enable'], $parameters['parameters']['ps_caching'])
+        && true === $parameters['parameters']['ps_cache_enable']
+        && isset($cacheType[$parameters['parameters']['ps_caching']])
+    ) {
+        foreach ($cacheType[$parameters['parameters']['ps_caching']] as $type) {
```

Comment:
So if memcache and apcu and xcache are available, we decide to use memcache? there's a "ranking"?

Reply:
@matks We use the parameter key `ps_caching` to know what kind of cache to use